config/plugins/pbh.py

```diff
@@ -0,0 +1,852 @@
+
+        hash_field = hash_field_obj['hash_field']
+
+        ip_mask_hash_field_correspondence_validator(ip_mask, hash_field)
+
+
+def interfaces_list_validator(db, interface_list, is_update):
+    if is_update and (interface_list is None):
+        return
+
+    is_error = False
+    interfaces_splited = interface_list.split(',')
+
+    for intf in interfaces_splited:
+        if intf.startswith('Ethernet'):
+            if not clicommon.is_valid_port(db.cfgdb, intf):
+                is_error = True
+                break
+        elif intf.startswith('PortChannel'):
+            if not clicommon.is_valid_portchannel(db.cfgdb, intf):
+                is_error = True
+                break
+        else:
+            is_error = True
+            break
+
+    if is_error:
+        exit_with_error("Error: invalid value '{}', for '--interface-list' option".format(interface_list), fg="red")
+
+
+@click.group(
+    name='pbh',
+    cls=clicommon.AliasedGroup
+)
+def PBH():
+    """ Configure PBH (Policy based hashing) feature """
+
+    pass
+
+
+@PBH.group(
+    name="hash-field",
+    cls=clicommon.AliasedGroup
+)
+def PBH_HASH_FIELD():
+    """ Configure PBH hash field """
+
+    pass
+
+
+@PBH_HASH_FIELD.command(name="add")
+@click.argument(
+    "hash-field-name",
+    nargs=1,
+    required=True,
+)
+@click.option(
+    "--hash-field",
+    help="Configure native hash field for this hash field",
+    required=True,
+    type=click.Choice(hash_field_types)
+)
+@click.option(
+    "--ip-mask",
+    help="""Configures IPv4/IPv6 address mask for this hash field, required when the value of --hash-field are - INNER_DST_IPV4 or
+    INNER_SRC_IPV4 or INNER_SRC_IPV4 or INNER_SRC_IPV4 """,
+    callback=ip_address_validator,
+)
+@click.option(
+    "--sequence-id",
+    help="Configures in which order the fields are hashed and defines which fields should be associative",
+    required=True,
+    type=click.INT,
+)
+@clicommon.pass_db
+def PBH_HASH_FIELD_add(db, hash_field_name, hash_field, ip_mask, sequence_id):
+    """ Add object to PBH_HASH_FIELD table """
+
+    ip_mask_hash_field_correspondence_validator(ip_mask, hash_field)
+
+    table = pbh_hash_field_tbl_name
+    key = hash_field_name
+    data = {}
+    if hash_field is not None:
+        data["hash_field"] = hash_field
+    if ip_mask is not None:
+        data["ip_mask"] = ip_mask
+    if sequence_id is not None:
+        data["sequence_id"] = sequence_id
+
+    try:
+        add_entry(db.cfgdb, table, key, data)
+    except Exception as err:
+        exit_with_error("Error: {}".format(err), fg="red")
+
+
+@PBH_HASH_FIELD.command(name="update")
+@click.argument(
+    "hash-field-name",
+    nargs=1,
+    required=True,
+)
+@click.option(
+    "--hash-field",
+    help="Configures native hash field for this hash field",
+    type=click.Choice(hash_field_types)
+)
+@click.option(
+    "--ip-mask",
+    help="Configures IPv4/IPv6 address mask for this hash field",
```

Comment:
@vadymhlushko-mlnx please align the description with add subgroup. Use YANG model as a reference

Reply:
@nazariig it is already according to the [YANG model](https://github.com/Azure/sonic-buildimage/pull/7461)

Reply:
@vadymhlushko-mlnx please rework as per internal review comments